### PR TITLE
Added Exponent XDE.app v2.8.0

### LINKED ISSUE #24295: Cask request: Exponent Development Environment (XDE)
### Cask details

**Name** - Exponent Development Environment (XDE)

**Homepage** - https://getexponent.com/

**License** - MIT - https://github.com/exponentjs/xde/blob/master/LICENSE

**Download URL** - https://xde-updates.exponentjs.com/download/mac

**Description** - Exponent is a mobile app development tool that enables you to build high quality native iOS and Android applications using JavaScript and deploy them effortlessly.

### PR DESCRIPTION
**Pull request by adamo57**
### Checklist
- [x] The commit message includes the cask’s name and version.
- [x] `brew cask audit --download exponent-xde` is error-free.
- [x] `brew cask style --fix exponent-xde` left no offenses.
- [x] Checked there are no open [pull requests](https://github.com/caskroom/homebrew-cask/pulls) for the same cask.
- [x] Checked there are no closed [issues](https://github.com/caskroom/homebrew-cask/issues) where that cask was already refused.
- [x] When naming the cask, followed the [token reference](https://github.com/caskroom/homebrew-cask/blob/master/doc/cask_language_reference/token_reference.md).
- [x] `brew cask install exponent-xde` worked successfully.
- [x] `brew cask uninstall exponent-xde` worked successfully.

Closes #24295.

### CODE REVIEW

File: Casks/exponent-xde.rb [hunk 1]
```diff
@@ -2,8 +2,8 @@
   version '2.8.0'
   sha256 'a9eb36f75a9f971afc923e0866b7359e33d7409bbfdfa9e5715d0213d7749ff8'
 
-  # https://xde-updates.exponentjs.com/download/mac was verified as official when first introduced to the cask
-  url 'https://xde-updates.exponentjs.com/download/mac'
+  # xde-updates.exponentjs.com/download/mac was verified as official when first introduced to the cask
+  url 'xde-updates.exponentjs.com/download/mac'
```
Comment (reitermarkus) [inline Casks/exponent-xde.rb]: Leave the URL scheme here in place.

File: Casks/exponent-xde.rb [hunk 2]
```diff
@@ -2,8 +2,8 @@
   version '2.8.0'
   sha256 'a9eb36f75a9f971afc923e0866b7359e33d7409bbfdfa9e5715d0213d7749ff8'
 
-  # https://xde-updates.exponentjs.com/download/mac was verified as official when first introduced to the cask
-  url 'https://xde-updates.exponentjs.com/download/mac'
+  # xde-updates.exponentjs.com/download/mac was verified as official when first introduced to the cask
```
Comment (reitermarkus) [inline Casks/exponent-xde.rb]: `xde-updates.exponentjs.com` is enough here, maybe even `exponentjs.com`, as it seems to belong to the same vendor.

**Review by reitermarkus — CHANGES_REQUESTED** [review]

File: Casks/exponent-xde.rb [hunk 3]
```diff
@@ -0,0 +1,12 @@
+cask 'exponent-xde' do
+  version '2.8.0'
+  sha256 'a9eb36f75a9f971afc923e0866b7359e33d7409bbfdfa9e5715d0213d7749ff8'
+
+  # exponentjs.com/download/mac was verified as official when first introduced to the cask
```
Comment (reitermarkus) [inline Casks/exponent-xde.rb]: Only the [identifying part of the URL](https://github.com/caskroom/homebrew-cask/blob/master/doc/cask_language_reference/stanzas/url.md#when-url-and-homepage-hostnames-differ-add-a-comment) is needed for this comment, so you can strip `/download/mac`.

**Review by reitermarkus — CHANGES_REQUESTED** [review]

File: Casks/exponent-xde.rb [hunk 4]
```diff
@@ -0,0 +1,12 @@
+cask 'exponent-xde' do
+  version '2.8.0'
+  sha256 'a9eb36f75a9f971afc923e0866b7359e33d7409bbfdfa9e5715d0213d7749ff8'
```
Comment (reitermarkus) [inline Casks/exponent-xde.rb]: Since the URL is unversioned, use:

```ruby
version :latest
sha256 :no_check
```

**Review by reitermarkus — CHANGES_REQUESTED** [review]

**Review by reitermarkus — APPROVED** [review]

**Comment by reitermarkus** [comment]
Thanks!

**Comment by radixhound** [comment]
If anyone stumbles upon this issue and tries to use the info here, the cask is now called `expo-xde`

**Pull request merged**